nemo/collections/asr/data/data_simulation.py

```diff
@@ -511,78 +541,96 @@ def _add_file(
+        start_cutoff, start_window_amount = self._get_start_buffer_and_window(first_alignment)
+        if not self._params.data_simulator.session_params.start_window:  # cut off the start of the sentence
+            start_window_amount = 0
 
         # ensure the desired number of words are added and the length of the output session isn't exceeded
         sentence_duration_sr = len(self._sentence)
+
         remaining_duration_sr = max_sentence_duration_sr - sentence_duration_sr
         remaining_duration = max_sentence_duration - sentence_duration
         prev_dur_sr, dur_sr = 0, 0
-        nw, i = 0, 0
-        while nw < remaining_duration and dur_sr < remaining_duration_sr and i < len(file['words']):
-            dur_sr = int(file['alignments'][i] * self._params.data_simulator.sr) - start_cutoff
+        nw = 0
+        word_idx = offset_idx
+        silence_count = 1
+        while nw < remaining_duration and dur_sr < remaining_duration_sr and word_idx < len(file['words']):
+            dur_sr = int(file['alignments'][word_idx] * self._params.data_simulator.sr) - start_cutoff
+
             if dur_sr > remaining_duration_sr:
                 break
 
-            word = file['words'][i]
+            word = file['words'][word_idx]
+
+            if silence_count > 0 and word == "":
+                break
+
             self._words.append(word)
             self._alignments.append(
                 float(sentence_duration_sr * 1.0 / self._params.data_simulator.sr)
                 - float(start_cutoff * 1.0 / self._params.data_simulator.sr)
-                + file['alignments'][i]
+                + file['alignments'][word_idx]
             )
 
             if word == "":
-                i += 1
+                word_idx += 1
+                silence_count += 1
                 continue
             elif self._text == "":
                 self._text += word
             else:
                 self._text += " " + word
-            i += 1
+
+            word_idx += 1
             nw += 1
             prev_dur_sr = dur_sr
 
         # add audio clip up to the final alignment
-        if (
-            sentence_duration == 0
-        ) and self._params.data_simulator.session_params.window_type != None:  # cut off the start of the sentence
+        if self._params.data_simulator.session_params.window_type != None:  # cut off the start of the sentence
             if start_window_amount > 0:  # include window
                 window = self._get_window(start_window_amount, start=True)
+                self._sentence = self._sentence.to(self._device)
                 self._sentence = torch.cat(
                     (
                         self._sentence,
-                        np.multiply(audio_file[start_cutoff : start_cutoff + start_window_amount], window),
+                        # np.multiply(audio_file[start_cutoff : start_cutoff + start_window_amount], window),
```

Comment:
Let's remove the line that's commented out.

Reply:
Removed